src/LayerPlan.cpp

```diff
@@ -921,26 +923,56 @@ void LayerPlan::addWall(const LineJunctions& wall, int start_idx, const Settings
     for(size_t point_idx = 1; point_idx < max_index; point_idx++)
     {
         const ExtrusionJunction& p1 = wall[(wall.size() + start_idx + point_idx * direction) % wall.size()];
-        const coord_t line_width = (p0.w + p1.w) / 2; //For lines which in itself vary in width, use the average width of the variable line.
 
-        if (!bridge_wall_mask.empty())
+        if(!bridge_wall_mask.empty())
         {
             computeDistanceToBridgeStart((wall.size() + start_idx + point_idx * direction - 1) % wall.size());
         }
 
-        if (first_line)
+        if(first_line)
         {
             addTravel(p0.p, always_retract);
             first_line = false;
         }
-        if (is_small_feature)
+
+        /*
+        If the line has variable width, break it up into pieces with the
+        following constraints:
+        - Each piece must be smaller than the Maximum Resolution setting.
+        - The difference between the trapezoidal shape of the line and the
+          rectangular shape of the line may not exceed the Maximum Extrusion
+          Area Deviation setting, unless required by the first constraint.
+        Since breaking up a line segment into N pieces (each with averaged
+        width) divides the area deviation by N, we can simply check how many
+        pieces we'd want to get low enough deviation, then check if each piece
+        is not too short at the end.
+        */
+        const coord_t delta_line_width = p1.w - p0.w;
+        const Point line_vector = p1.p - p0.p;
+        const coord_t line_length = vSize(line_vector);
+        const coord_t line_area_deviation = std::abs(delta_line_width) * line_length / 8; //How much the line would deviate from the trapezoidal shape if printed at average width.
```

Comment:
When I calculate the area of the deviation I end up with 
```cpp
const coord_t line_area_deviation = std::abs(delta_line_width) * line_length / 4;


```
My reasoning:
![Untitled Diagram drawio](https://user-images.githubusercontent.com/6638028/147266956-9e4a2d59-b764-498e-b7bd-7e6dd477e6a6.png)

Reply:
Actually, I believe we don't even need to calculate the deviation area. We can directly calculate how much the area deviates if the line is partitioned in `n` parts.

If a line is partitioned in `n` parts then `4n` of these deviation triangles are created. Each of this triangles have corresponding shapes with a width that is equal to `line_width / n` and a height that is equal to `abs(delta_width) / 4n` and a width that is equal to `line_length / n`. The area such a triangle is equal to 
```
triangle_area_deviation = (w * h) / 2 
                        = ((abs(delta_width) / 4n) * (line_length / n)) / 2 
                        = (abs(delta_width) * line_length) / (8n^2)
```
Multiply this by the number of triangles, `4n`, we get 
```
total_area_deviation = (abs(delta_width) * line_length) / (8n^2) * 4n 
                     = (abs(delta_width) * line_length) / (4n)
```
So if we partition a line into `n` parts the total area deviation will be `(abs(delta_width) * line_length) / (4n)`, Solve for `n`
```
total_area_deviation      = (abs(delta_width) * line_length) / (4n)
total_area_deviation * 4n = abs(delta_width) * line_length
                        n = (abs(delta_width) * line_length) / (total_area_deviation * 4)
```
Now we can compute how many pieces we need if we want to bound the total area deviation by `max_area_deviation` by simply substituting the variable.
```cpp
size_t pieces = round_up_divide(abs(delta_width) * line_length, max_area_deviation * 4)
```

Reply:
Yes, indeed I did something similar on a piece of paper when I got that formula. Maybe I mistakenly divided the length by 2 as well back then or something. I'm now arriving at `/ 4` as well.